code/datums/wires/jukebox.dm

```diff
@@ -0,0 +1,86 @@
+/datum/wires/jukebox
+	holder_type = /obj/machinery/media/jukebox
+	wire_count = 8
+	var/interference = 0 //Caused by pulsing the transmit wire
+
+var/const/JUKE_POWER_ONE = 1 //Power. Cut for shock and off. Pulse toggles.
+var/const/JUKE_POWER_TWO = 2 //Power. Cut for shock and off. Pulse toggles increased lum.
+var/const/JUKE_POWER_THREE = 4 //Power. Cut for shock and off. Pulse toggles decreased lum.
+var/const/JUKE_SHUFFLE = 8 //Cut to disable shuffle and move to play_once. Pulse immediately shuffles.
+var/const/JUKE_CAPITAL = 16 //Cut to disable song picking. Pulse randomizes song pick price 1-10.
+var/const/JUKE_TRANSMIT = 32 //Cut shocks and disables multitool. Pulse emits burst of rads.
+var/const/JUKE_CONFIG = 64 //Cut emags. Pulse plays IAMERROR.ogg
+var/const/JUKE_SETTING = 128 //Cut shocks. Pulse toggles settings menu.
+
+/datum/wires/jukebox/CanUse(var/mob/living/L)
+	var/obj/machinery/media/jukebox/J = holder
+	if(J.panel_open)
+		return 1
+	return 0
+
+/datum/wires/jukebox/GetInteractWindow()
+	var/obj/machinery/media/jukebox/J = holder
+	. += ..()
+	. += "<BR>The decorative tube with bubbles is [!J.any_power_cut() ? "glowing" : "dim"].<BR>"
+	. += "The green slider bar is [!IsIndexCut(JUKE_TRANSMIT) ? "modulating around full" : "empty"].<BR>"
+	. += "The maintenance button is [J.access_unlocked ? "lit" : "off"].<BR>"
+	. += "An unlabelled light is [J.emagged ? "dark" : "blinking occasionally"].<BR>"
+
+/datum/wires/jukebox/UpdatePulsed(var/index)
+	if(interference)
+	var/obj/machinery/media/jukebox/J = holder
+	switch(index)
+		if(JUKE_POWER_ONE||JUKE_POWER_TWO||JUKE_POWER_THREE)

```

Comment:
This doesn't work.
It's 
`if(JUKE_POWER_ONE, JUKE_POWER_TWO, JUKE_POWER_THREE)`